src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -261,6 +261,9 @@ public EventBus unregisterDefaultCodec(Class clazz) {
   @Override
   public void close(Handler<AsyncResult<Void>> completionHandler) {
     unregisterAllHandlers();
+    if (metrics != null) {
+      metrics.close();
```

Comment:
I think we should just say in the documentation that metrics.close() can be called from any thread.

Then we don't need to run close on a context anywhere which will simplify things.